infrastructure/lib/atat-net-stack.ts

```diff
@@ -0,0 +1,114 @@
+    // });
+    const tgwAttachment = new ec2.CfnTransitGatewayAttachment(this, "VpcTgwAttachment", {
+      vpcId: vpc.vpcId,
+      subnetIds: vpc.isolatedSubnets.map((subnet) => (subnet as ec2.Subnet).subnetId),
+      transitGatewayId: transitGatewayId,
+    });
+    this.addDefaultTransitGatewayRoute(tgwAttachment);
+
+    const dynamodbEndpoint = vpc.addGatewayEndpoint("DynamodbEndpoint", {
+      service: ec2.GatewayVpcEndpointAwsService.DYNAMODB,
+    });
+    const s3Endpoint = vpc.addInterfaceEndpoint("S3Endpoint", {
+      // S3 Interface Endpoints are not currently available as a value under
+      // ec2.InterfaceVpcEndpointAwsService so we have to construct it manually.
+      service: { name: `com.amazonaws.${cdk.Aws.REGION}.s3`, port: 443, privateDnsDefault: false },
+    });
+    const apiGatewayEndpoint = vpc.addInterfaceEndpoint("ApiGatewayEndpoint", {
+      service: ec2.InterfaceVpcEndpointAwsService.APIGATEWAY,
+    });
+    const lambdaEndpoint = vpc.addInterfaceEndpoint("LambdaEndpoint", {
+      service: ec2.InterfaceVpcEndpointAwsService.LAMBDA,
+    });
+    const logsEndpoint = vpc.addInterfaceEndpoint("LogsEndpoint", {
+      service: ec2.InterfaceVpcEndpointAwsService.CLOUDWATCH_LOGS,
+    });
+    const sqsEndpoint = vpc.addInterfaceEndpoint("SqsEndpoint", {
+      service: ec2.InterfaceVpcEndpointAwsService.SQS,
+    });
+    this.endpoints.push(dynamodbEndpoint, s3Endpoint, apiGatewayEndpoint, lambdaEndpoint, logsEndpoint, sqsEndpoint);
+  }
+
+  private addDefaultTransitGatewayRoute(transitGatewayAttachment: ec2.CfnTransitGatewayAttachment) {
+    const subnets = this.vpc.isolatedSubnets as ec2.Subnet[];
+    subnets.forEach((subnet, index) => {
+      const table = subnet.routeTable.routeTableId;
+      // The CDK does not support adding a Transit Gateway Route to a subnet directly
+      // (as of v1.125.0) so we have to create the CfnRoute manually. There is not a
+      // L2 construct for Routes.
+      const route = new ec2.CfnRoute(this, `Subnet${index + 1}DefaultRouteToTgw`, {
+        routeTableId: table,
+        destinationCidrBlock: "0.0.0.0/0",
+        transitGatewayId: transitGatewayAttachment.transitGatewayId,
+      });
+      // Because only the TransitGatewayId gets passed to the CfnRoute, we may not
+      // necessarily have the implicit dependency that we'd expect. If the route
+      // starts Creation before the attachment is accepted and attached then the
+      // route will fail to be created and the stack creation will fail.
+      route.addDependsOn(transitGatewayAttachment);
+      // We need to disable this check because we're actually bypassing the
+      // private member check in TypeScript. This is because we may want to leverage
+      // the actual `internetConnectivityEstablished` attribute later
+      // eslint-disable-next-line dot-notation
+      subnet["_internetConnectivityEstablished"].add(route);
+    });
+  }
+
+  private findTransitGateway(): string {
```

Comment:
Always happy to see more verbosity than less, so much appreciated. And I think this seems totally reasonable for starting up. We can work with Kyle Loving to define this hand-off (e.g. the SM PS param) but I wouldn't want to hold this work up while we wait on that.

Reply:
And since it's a separate function we can change the internal implementation as needed so long as the actual underlying Transit Gateway ID doesn't change (which it shouldn't)